homeassistant/components/binary_sensor/velbus.py

```diff
@@ -4,82 +4,58 @@
 For more details about this platform, please refer to the documentation at
 https://home-assistant.io/components/binary_sensor.velbus/
 """
-import asyncio
 import logging
 
-
-import voluptuous as vol
-
-from homeassistant.const import CONF_NAME, CONF_DEVICES
 from homeassistant.components.binary_sensor import BinarySensorDevice
-from homeassistant.components.binary_sensor import PLATFORM_SCHEMA
 from homeassistant.components.velbus import DOMAIN
-import homeassistant.helpers.config_validation as cv
-
-
-DEPENDENCIES = ['velbus']
 
 _LOGGER = logging.getLogger(__name__)
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
-    vol.Required(CONF_DEVICES): vol.All(cv.ensure_list, [
-        {
-            vol.Required('module'): cv.positive_int,
-            vol.Required('channel'): cv.positive_int,
-            vol.Required(CONF_NAME): cv.string,
-            vol.Optional('is_pushbutton'): cv.boolean
-        }
-    ])
-})
+DEPENDENCIES = ['velbus']
 
 
-def setup_platform(hass, config, add_devices, discovery_info=None):
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
     """Set up Velbus binary sensors."""
-    velbus = hass.data[DOMAIN]
-
-    add_devices(VelbusBinarySensor(sensor, velbus)
-                for sensor in config[CONF_DEVICES])
+    if discovery_info is None:
+        return
+    sensors = []
+    for sensor in discovery_info:
+        module = hass.data[DOMAIN].get_module(sensor[0])
+        channel = sensor[1]
+        sensors.append(VelbusBinarySensor(module, channel))
+    async_add_devices(sensors, update_before_add=True)
 
 
 class VelbusBinarySensor(BinarySensorDevice):
     """Representation of a Velbus Binary Sensor."""
 
-    def __init__(self, binary_sensor, velbus):
+    def __init__(self, module, channel):
```

Comment:
Break out the common methods and properties into a common base velbus device entity class defined in the velbus component. Then use multiple inheritance in the platform entities, when defining those classes.